File: backend/src/main/java/com/smsgateway/model/dto/DeviceTrend.java
package com.smsgateway.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * 设备端主页的两张小图：近 N 天 + 今日逐小时。
 *
 * 合成一个响应而不是两个端点：这两张图永远一起出现、刷新节奏也一样，
 * 分成两个接口就是每次刷新多一个往返 —— 而这台设备是 7×24 挂着的。
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DeviceTrend {

    /** 近 N 天，按天升序，**缺的天已补 0**（图里要有空柱子，不能跳过去）。 */
    private List<DailyCount> daily;

    /** 今日 0-23 点，**24 个点一个不少，没有数据的小时补 0**。 */
    private List<HourlyCount> hourly;
}

File: android/app/src/main/java/com/smsgateway/app/model/DeviceTrend.kt
package com.smsgateway.app.model

/**
 * 主页那两张小图的数据：近 N 天 + 今日逐小时。
 *
 * 全部给默认值：Gson 遇到缺字段时不会因此抛异常（与 SmsRecord 同一套写法）。
 * 缺的天/小时由**服务端**补 0 —— 柱状图里的空柱子本身就是信息（那天一条都没收到），
 * 客户端再补一遍就等于两处各有一套「补零规则」。
 */
data class DeviceTrend(
    val daily: List<DailyStat> = emptyList(),
    val hourly: List<HourlyStat> = emptyList()
)

/** 单日：日期（yyyy-MM-dd）+ 当天短信数 + 其中识别出验证码的条数。 */
data class DailyStat(
    val day: String = "",
    val value: Long = 0,
    val codes: Long = 0
)

/** 单个小时（0-23）。 */
data class HourlyStat(
    val hour: Int = 0,
    val value: Long = 0,
    val codes: Long = 0
)
